### PR TITLE
List: raise onClick when enter is pressed

### PR DESCRIPTION
**Pull request by AlexanderMoiseev**

### CODE REVIEW

File: js/ui/collection/ui.collection_widget.base.js [hunk 1]
```diff
@@ -61,6 +61,13 @@ const CollectionWidget = Widget.inherit({
                 return;
             }
 
+            const itemData = this._getItemData($itemElement);
+            if(itemData && itemData.onClick) {
```
Comment (ksercs) [inline js/ui/collection/ui.collection_widget.base.js]:
```suggestion
            if(itemData?.onClick) {
```

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 2]
```diff
@@ -1995,6 +1995,28 @@ QUnit.module('events', moduleSetup, () => {
         assert.strictEqual(actionData.itemIndex.group, 1, 'correct groupIndex passed');
     });
 
+    QUnit.test('onClick for item should be fired on \'enter\' key pressed', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js]:
```suggestion
    QUnit.test('item onClick handler should be fired on "enter" key press', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 3]
```diff
@@ -1995,6 +1995,28 @@ QUnit.module('events', moduleSetup, () => {
...
+            return;
+        }
+
+        const $element = this.element.dxList({
+            items:[{
+                text: 'item 1',
+                onClick: () => {
+                    assert.ok('onClick was fired');
+                }
+              }
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js]: looks like missformatting
```suggestion
            }]
```

File: testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 4]
```diff
@@ -1995,6 +1995,28 @@ QUnit.module('events', moduleSetup, () => {
...
+            ]
+        });
+
+        const $item = $element.find(toSelector(LIST_ITEM_CLASS));
+
+        $item.trigger('dxpointerdown');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:2014]: why do we do this? For focusing? If yes, we can write it more intuitive way
```suggestion
        $item.focus();
```

Comment (AlexanderMoiseev) [reply testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:2014]: Yes for focusing, but $item.focus() doesn't work, may be dxpointerdown used in all other places in this file for a reason

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Review by marker-dao — APPROVED** [review]

**Pull request merged**